.travis.yml

```diff
@@ -12,35 +12,47 @@ matrix:
-          - sudo apt-get update
-          - sudo apt-get install g++-7
-          - export BOOST_URL="https://downloads.sourceforge.net/project/boost/boost/1.67.0/boost_1_67_0.tar.gz"
-          - mkdir -p /tmp/boost
-          - wget --quiet -O - ${BOOST_URL} | tar --strip-components=1 -xz -C /tmp/boost  || exit 1
-          - cd /tmp/boost/tools/build && ./bootstrap.sh && ./b2 install --prefix=/tmp/b2
-          - export PATH=/tmp/b2/bin:${PATH}
-          - cd /tmp/boost && b2 toolset=gcc cxxflags="-std=c++14 ${CXXFLAGS}"
+        # C++17
+        - sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
+        - sudo apt-get update -qq
+      install:
+        # C++17
+        - sudo apt-get install -qq g++-8 libboost-all-dev
+        - sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-8 90
+      after_success:
+        # Create lcov report
+        # capture coverage info
+        - lcov --directory . --capture --output-file coverage.info
+        # filter out system and extra files.
+        # To also not include test code in coverage add them with full path to the patterns: '*/tests/*'
+        - lcov --remove coverage.info '/usr/*' "${HOME}"'/.cache/*' --output-file coverage.info
+        # output coverage data for debugging (optional)
+        - lcov --list coverage.info
+        # Uploading to CodeCov
+        # '-f' specifies file(s) to use and disables manual coverage gathering and file search which has already been done above
+        - bash <(curl -s https://codecov.io/bash) -f coverage.info || echo "Codecov did not collect coverage reports"
     - os: osx
-      osx_image: xcode9.4
-      env:
+      osx_image: xcode10
       before_install:
-          - brew install lcov
+        - brew install lcov
       after_success:
-          # Creating report
-          - cd ${TRAVIS_BUILD_DIR}
-          - lcov --directory . --capture --output-file coverage.info # capture coverage info
-          - lcov --remove coverage.info '/usr/*' --output-file coverage.info # filter out system
-          - lcov --list coverage.info #debug info
-          # Uploading report to CodeCov
-          - bash <(curl -s https://codecov.io/bash) || echo "Codecov did not collect coverage reports"
+        # Creating report
+        - cd ${TRAVIS_BUILD_DIR}
+        - lcov --directory . --capture --output-file coverage.info # capture coverage info
+        - lcov --remove coverage.info '/usr/*' --output-file coverage.info # filter out system
+        - lcov --list coverage.info #debug info
+        # Uploading report to CodeCov
+        - bash <(curl -s https://codecov.io/bash) || echo "Codecov did not collect coverage reports"
 
   allow_failures:
-          - os: linux
+    - os: osx
 
 script:
-  - cmake -DCMAKE_BUILD_TYPE=Debug -G "Unix Makefiles" . && make && make test  
+  - cmake -DCMAKE_BUILD_TYPE=Debug -DREAL_BENCH=OFF -G "Unix Makefiles" . && make -j`nproc` && make -j`nproc` test  
```

Comment:
What is that REAL_BENCH for?

Reply:
It is an artifact left from my importation of BoostReal files, I didn't check thoroughly to remove what doesn't makes sense here